app/src/main/java/org/gdg/frisbee/android/pulse/PulseFragment.java

```diff
@@ -162,10 +164,23 @@ public void networkFailure(Throwable error) {
         }
     }
 
-    private void initAdapter(Pulse pulse) {
-        mAdapter.setPulse(mMode, pulse);
-        mAdapter.notifyDataSetChanged();
-        setIsLoading(false);
+    private void initAdapter(final Pulse pulse) {
```

Comment:
You are getting the chapter list after everything is ready. If we always need the chapter list, what about getting it upfront? 
First do the chapter retrieval and then do the rest. 
Then we can provide the Directory with the constructor of the adapter. Then we can make it private final. 

What do you think?